mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/validator/IdentityDataValidator.java

```diff
@@ -0,0 +1,54 @@
+package io.mosip.esignet.mock.identitysystem.validator;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.mosip.esignet.mock.identitysystem.dto.IdentityData;
+
+@Component
+public class IdentityDataValidator implements ConstraintValidator<IdData, IdentityData> {
+
+	@Value("${mosip.esignet.mock.supported-fields}")
+	private String supportedFields;
+
+	@Value("${mosip.esignet.mock.number-of-extra-fields:10}")
+	private String extraFields;

```

Comment:
The requirement on extraFields was allowing someone to include their custom fields while creating the identity, how are we handling that if that fileds not declared in the Identity data class ?

Reply:
Modified the approach as discussed.